data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/actions/CombineAggregateActionTest.java

```diff
@@ -0,0 +1,109 @@
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class CombineAggregateActionTest {
+    private AggregateAction combineAggregateAction;
+    private List<Event> events;
+    private List<Map<String, Object>> eventMaps;
+
+    @BeforeEach
+    void setup() {
+        events = new ArrayList<>();
+        eventMaps = new ArrayList<>();
+        combineAggregateAction = new CombineAggregateAction();
+
+        final Map<String, Object> firstEventMap = new HashMap<>();
+        firstEventMap.put("string", "firstEventString");
+        firstEventMap.put("array", Arrays.asList(1, 2, 3));
+        eventMaps.add(firstEventMap);
+        events.add(buildEventFromMap(firstEventMap));
+
+        final Map<String, Object> secondEventMap = new HashMap<>();
+        secondEventMap.put("string", "secondEventString");
+        secondEventMap.put("number", 2);
+        eventMaps.add(secondEventMap);
+        events.add(buildEventFromMap(secondEventMap));
+
+        final Map<String, Object> thirdEventMap = new HashMap<>();
+        thirdEventMap.put("string", "thirdEventString");
+        thirdEventMap.put("double", 2.1);
+        eventMaps.add(thirdEventMap);
+        events.add(buildEventFromMap(thirdEventMap));
+
+    }
+
+    private Event buildEventFromMap(final Map<String, Object> eventMap) {
+        return JacksonEvent.builder()
+                .withEventType("event")
+                .withData(eventMap)
+                .build();
+    }
+
+    @Test
+    void handleEvent_with_null_group_state_should_return_correct_AggregateResponse() {
+        final AggregateActionResponse aggregateActionResponse = combineAggregateAction.handleEvent(events.get(0), null);
+
+        assertThat(aggregateActionResponse.getEvent().isPresent(), equalTo(false));
+        assertThat(aggregateActionResponse.isCloseWindowNow(), equalTo(false));
+    }
+
+    @Test
+    void combining_with_single_event_should_result_in_that_same_event_being_returned_on_concludeGroup() {
+        final Map<Object, Object> groupState = new HashMap<>();
+        final AggregateActionResponse aggregateActionResponse = combineAggregateAction.handleEvent(events.get(0), groupState);
+        assertThat(aggregateActionResponse.getEvent().isPresent(), equalTo(false));
+        assertThat(aggregateActionResponse.isCloseWindowNow(), equalTo(false));
+
+        final Optional<Event> result = combineAggregateAction.concludeGroup(groupState);
```

Comment:
I recommend that you only test one function per test. If any assertions fail after the second call, it is difficult to tell which function was in error. This approach can also reduce the number of complexity of test cases.

The best solution will be to change the `handleEvent` method to make assertions on the `groupState`. Then in the test for `concludeGroup`, setup the state correctly.